jenkins/init.sls

```diff
@@ -22,25 +15,34 @@ jenkins_user:
     - home: {{ jenkins.home }}
     - shell: /bin/bash
     - require:
-      - group: jenkins_group
+      - group: create_jenkins_group
+  file.directory:
+    - name: {{ jenkins.home }}
+    - user: {{ jenkins.user }}
+    - group: {{ jenkins.group }}
+    - mode: 755
+    - require:
+      - user: create_jenkins_user
 
 jenkins:
   {% if grains['os_family'] in ['RedHat', 'Debian'] %}
   pkgrepo.managed:
     - humanname: Jenkins upstream package repository
     {% if grains['os_family'] == 'RedHat' %}
-    - baseurl: http://pkg.jenkins-ci.org/redhat
-    - gpgkey: http://pkg.jenkins-ci.org/redhat/jenkins-ci.org.key
+    - baseurl: http://pkg.jenkins-ci.org/redhat{{ jenkins.repo_version }}
+    - gpgkey: http://pkg.jenkins-ci.org/redhat{{ jenkins.repo_version }}/jenkins-ci.org.key
     {% elif grains['os_family'] == 'Debian' %}
     - file: {{jenkins.deb_apt_source}}
     - name: deb http://pkg.jenkins-ci.org/debian binary/
     - key_url: http://pkg.jenkins-ci.org/debian/jenkins-ci.org.key
     {% endif %}
-    - require_in:
-      - pkg: jenkins
+    - require:
+      - file: create_jenkins_user
```

Comment:
Because the creation of jenkins user is independent from the installation of jenkins package and would not cause any problems if that is done before the latter action. I just thought it would be better to first have a jenkins group and user before installing the package. Do you think it is not necessary?